cookbooks/cdo-apps/templates/default/unicorn.sh.erb

```diff
@@ -15,6 +15,8 @@ PID=<%= @pid_file %>
 CMD="cd $APP_ROOT; LANG=en_US.UTF-8 bundle exec unicorn -D -c <%= @src_file %> -E <%= @env %>"
 AS_USER=<%= @user %>
 <%= @bundle_env ? @bundle_env.map{|k,v|"export #{k}=#{v}"}.join("\n") : '' %>
+UPGRADE_DELAY=${UPGRADE_DELAY-4}
```

Comment:
This line comes from [upstream](http://repo.or.cz/unicorn.git/blob/HEAD:/examples/init.sh#l21); `parameter=${parameter-default}` is [Bash parameter substitution](http://www.tldp.org/LDP/abs/html/parameter-substitution.html) syntax for setting a parameter default but using the existing value if already set (roughly equivalent to e.g., `parameter = parameter || default` in Ruby).

The magic number `4` for the default upgrade delay is a little arbitrary- my previous setting of `1` second didn't turn out to be quite long enough, because in testing I encountered an issue where the old-master PID had been renamed but the new-master PID hadn't yet been created, so the `upgrade` target exited with a `pid not found` error, causing the frontend provisioning to fail.

Although upstream had `2` seconds as a default, I wanted to be a little more sure our service didn't hit the same issue again so thought `4` would be a little safer. (The full upgrade process takes much longer than 4 seconds anyway, so there's little downside to increasing this number by a few seconds in our case.)